hydrolib/core/dflowfm/research/models.py

```diff
@@ -887,6 +888,45 @@ class Comments(Processes.Comments):
     )
 
 
+class ResearchSedTrails(INIBasedModel):
+    class Comments(INIBasedModel.Comments):
+        research_sedtrailsgrid: Optional[str] = Field(
+            "Grid file for sedtrails output locations on corners.",
+            alias="sedtrailsgrid",
+        )
+        research_sedtrailsanalysis: Optional[str] = Field(
+            "Sedtrails analysis. Should be all, transport, flowvelocity or soulsby.",
+            alias="sedtrailsanalysis",
+        )
+        research_sedtrailsinterval: Optional[str] = Field(
+            "Sedtrails output times (s), interval, starttime, stoptime (s), if starttime, stoptime are left blank, use whole simulation period.",
+            alias="sedtrailsinterval",
+        )
+        research_sedtrailsoutputfile: Optional[str] = Field(
+            "Sedtrails time-avgd output file.", alias="sedtrailsoutputfile"
+        )
+
+    comments: Comments = Comments()
+    _header: Literal["sedtrails"] = "sedtrails"
+
+    research_sedtrailsgrid: Optional[DiskOnlyFileModel] = Field(
+        None, alias="sedtrailsgrid"
+    )
+    research_sedtrailsanalysis: Optional[
+        Literal["all", "transport", "flowvelocity", "soulsby"]
+    ] = Field(None, alias="sedtrailsanalysis")
+    research_sedtrailsinterval: Optional[List[float]] = Field(
+        None, alias="sedtrailsinterval"
+    )
+    research_sedtrailsoutputfile: Optional[DiskOnlyFileModel] = Field(
+        None, alias="sedtrailsoutputfile"
```

Comment:
Qua capitals, denk ik dat de aliases conform de kernel code zouden moeten zijn:
- `SedtrailsGrid`
- `SedtrailsAnalysis`
- `SedtrailsInterval`
- `SedtrailsOutputFile`

Reply:
Helaas kan dat hier niet zo 1, 2, 3. Op de een of andere manier moet er een mapping gemaakt kunnen worden van wat er in de .mdu file staat (of de keys in je dictionary als je een model aanmaakt via een dictionary) naar de properties. Dat kan of via de naam van de property, of via de alias. We kunnen de mapping niet via de naam doen, omdat we alles prefixen met _research__. Dus we moeten de mapping doen via de alias. Het lijkt erop dat pydantic die mapping moet maken in lowercase. Ik had namelijk eerst ook de capitals gedaan zoals het in de kernel code staat, maar dan werkt het niet!

Als je dit toch een belangrijk punt vindt, wil je er dan een issue voor aanmaken? Dan kunnen we in een volgende sprint misschien iets voor bouwen om het toch voor elkaar te krijgen.

Reply:
Ah, betekent dit dat alle research keywords lowercase zijn in de mdu? Wat mij betreft is dat op zich prima (hooguit iets minder leesbaar), ik had me dit alleen nog niet gerealiseerd.

Reply:
Dat klopt. Alle research keywords worden momenteel lowercase weggeschreven naar de .mdu.